### PR TITLE
Adds IssuedCurrencyClient::unfreezeTrustLine

### PR DESCRIPTION
**Pull request by tedkalaw**
## High Level Overview of Change
This PR adds IssuedCurrencyClient::unfreezeTrustLine and a test.

### Context of Change
This is the opposite of https://github.com/xpring-eng/Xpring-JS/pull/851, and allows unfreezing a frozen trust line.

https://xrpl.org/freezes.html#enabling-or-disabling-individual-freeze

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (non-breaking change that only restructures code)
- [ ] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

## Before / After
`IssuedCurrencyClient` has `unfreezeTrustLine`

## Test Plan
CI, test

### CODE REVIEW

File: src/XRP/issued-currency-client.ts [hunk 1]
```diff
@@ -479,6 +479,32 @@ export default class IssuedCurrencyClient {
...
+   *
+   * @param accountToUnfreeze The X-Address of the account which to unfreeze a trust line.
+   * @param currencyName The currency of the trust line to unfreeze.
+   * @param wallet The wallet unfreezing the trust line.
+   */
+  public async unfreezeTrustLine(
+    accountToUnfreeze: string,
+    currencyName: string,
+    wallet: Wallet,
+  ): Promise<TransactionResult> {
```
Comment (mvadari) [inline src/XRP/issued-currency-client.ts:496]: Should there be an optional parameter of amount?

File: src/XRP/issued-currency-client.ts [hunk 2]
```diff
@@ -479,6 +479,32 @@ export default class IssuedCurrencyClient {
     )
   }
 
+  /**
+   * Unfreezes the trust line between this account (issuing account) and another account.
+   * Note that the trust line's limit is set to 0.
+   *
+   * @see https://xrpl.org/freezes.html#enabling-or-disabling-individual-freeze
+   *
+   * @param accountToUnfreeze The X-Address of the account which to unfreeze a trust line.
```
Comment (amiecorso) [inline src/XRP/issued-currency-client.ts:488]: same note here about this confusing name...  I'd also call this `trustlineAccount` and the second parameter `trustlineCurrencyName` or just `trustlineCurrency` ...

Comment (amiecorso) [inline src/XRP/issued-currency-client.ts:488]:
```suggestion
   * @param accountToUnfreeze The X-Address of the account involved in the trust line being unfrozen.
```

Comment (amiecorso) [reply src/XRP/issued-currency-client.ts:488]: lol

File: src/XRP/issued-currency-client.ts (hunk 1 above)
Comment (amiecorso) [reply src/XRP/issued-currency-client.ts:496]: I think that we can operate on the assumption that gateways will never be using trustlines with non-zero limits extended to a counterparty... so we can probably get await without an `amount` param.  However, if we find we need to change this in future we can just add the optional param as a non-breaking change so we can do less now.

**Review by amiecorso — APPROVED** [review]
approving now but check my notes about the names and doc

**Review by mvadari — APPROVED** [review]

File: src/XRP/issued-currency-client.ts (hunk 2 above)
Comment (tedkalaw) [reply src/XRP/issued-currency-client.ts:488]: got it, per our conversation off line i changed this to `trustLinePeerAccount`

Comment (tedkalaw) [reply src/XRP/issued-currency-client.ts:488]: got it

**Pull request merged**